api/v1alpha1/multiclusterservice_types.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MultiClusterService is the Schema for the multiclusterservice API
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiClusterServiceSpec   `json:"spec,omitempty"`
+	Status MultiClusterServiceStatus `json:"status,omitempty"`
+}
+
+// MultiClusterServiceSpec defines the desired state of MultiClusterService
+type MultiClusterServiceSpec struct {
+	// ClusterSelector identifies target clusters to manage services on.
+	// +optional
+	ClusterSelector metav1.LabelSelector `json:"clusterSelector,omitempty"`
+	// Tier sets the priority for the services defined in this spec.
+	// Higher value means lower priority and vice versa.
+	// In case of conflict with another object managing the service,
+	// the one with lower priority will get to deploy its services.
+	// +kubebuilder:default:=100
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Tier int32 `json:"tier,omitempty"`
+	// StopOnConflict specifies what to do in case of a conflict.
+	// E.g. If another object is already managing a service.
+	// By default the remaining services will be deployed even if conflict is detected.
+	// If set to true, the deployment will stop after encountering the first conflict.
+	// +optional
+	StopOnConflict bool `json:"stopOnConflict,omitempty"`
+	// Services is a list of services created via ServiceTemplates
+	// that could be installed on the target cluster.
+	// +optional
+	Services []ServiceSpec `json:"services,omitempty"`
+}
+
+// ServiceSpec represents a Service to be managed
+type ServiceSpec struct {
```

Comment:
Can we align the fields in all the introduced/updated structs to decrease memory usage during runtime? It's not very important at the moment, but having the only binary it could affect on a distance

For example, this struct might be decreased from 64 to 48 bytes, and the `internal/sveltos.ReconcileProfileOpts` might be decreased from 24 to 16.

Applicable elsewhere

Reply:
I believe alignment should be good now.